weld/src/main/java/org/jboss/as/weld/deployment/WeldDeployment.java

```diff
@@ -222,7 +223,7 @@ public Module getModule() {
     }
 
     @Override
-    public synchronized BeanDeploymentArchive getBeanDeploymentArchive(final Class<?> beanClass) {
+    public BeanDeploymentArchive getBeanDeploymentArchive(final Class<?> beanClass) {
```

Comment:
on line 240 (containsKey), you should check if the specified key is not null otherwise NullPointerException